### PR TITLE
Add preliminary stream type; Update equality checking

### PR DESCRIPTION
**Pull request by andyscott**
This adds a basic lazy stream type based on `Nu[ListF, ?]`. I also went ahead and started to fix up `Eq` instances.

This opens up the door for a _lot_ of cleanup. In particular, I'd like to organize all of the imports so that it's intuitive to use.

### CODE REVIEW

File: modules/core/src/main/scala/qq/droste/basis.scala [hunk 1]
```diff
@@ -10,8 +10,10 @@ import data.list._
 import util.newtypes._
 
 import cats.Eval
-import cats.{Foldable, Monad}
-import cats.kernel.{Monoid, Eq}
+import cats.Eq
+import cats.Foldable
+import cats.Monad
+import cats.Monoid
```
Comment (pepegar) [inline modules/core/src/main/scala/qq/droste/basis.scala]: Why this change?

File: modules/core/src/main/scala/qq/droste/package.scala [hunk 2]
```diff
@@ -74,4 +78,19 @@ object `package` {
     def apply[M[_], F[_], G[_], A](f: F[A] => M[G[A]]): TransM[M, F, G, A] = GTransM(f)
   }
 
+  type Delay[F[_], G[_]] = F ~> (F ∘ G)#λ
```
Comment (pepegar) [inline modules/core/src/main/scala/qq/droste/package.scala]: What's the difference between `Delay` and `∘`?

File: modules/core/src/main/scala/qq/droste/basis.scala (hunk 1 above)
Comment (andyscott) [reply modules/core/src/main/scala/qq/droste/basis.scala]: Just removing the `.kernel` from the imports. In the process I expanded the single line imports, simply because I tend to avoid them.

Comment (pepegar) [reply modules/core/src/main/scala/qq/droste/basis.scala]: I think we could put those rules in a `.scalafmt.conf` file to reach homogeneicity (I think I spelled it wrong :D )?

Comment (andyscott) [reply modules/core/src/main/scala/qq/droste/basis.scala]: Yeah that'd be nice! Were you volunteering to take on the effort? 😄

Comment (pepegar) [reply modules/core/src/main/scala/qq/droste/basis.scala]: https://github.com/andyscott/droste/pull/94 there we go!
